### ISSUE TITLE
Support for ibm_db Driver Version 3.2.4 in Sequelize 6 for DB2 LUW

### ISSUE THREAD

**Issue by nickhais** [sequelize/sequelize#17538]
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information.
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md).

### Describe the feature you'd like to see implemented

Sequelize currently supports only `ibm_db` version 2 for DB2 LUW according to the documentation. The latest version of the `ibm_db` driver is 3.2.4. I would like to request Sequelize version 6 to support the new `ibm_db` driver.

![Image](https://github.com/user-attachments/assets/c15098c5-94ba-48ad-b598-74de77abba41)

### Describe why you would like this feature to be added to Sequelize

The `ibm_db` driver has had updates and improvements in version 3.2.4, which could benefit projects using DB2 LUW. Ensuring compatibility with the latest driver allows Sequelize users to take advantage of these improvements.

### Is this feature dialect-specific?

- [x] Yes. This feature only applies to the DB2 dialect.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [x] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.

---

_Indicate your interest in the addition of this feature by adding the 👍 reaction. Comments such as "+1" will be removed._

**Comment by WikiRik** [comment]
I'd recommend using the v7 alphas instead. DB2 support has been improved there and has become less experimental than in v6.

But if the community can make a PR without breaking changes to allow support for ibm_db v3 we can review that

**Comment by nickhais** [comment]
Thank you @WikiRik for the recommendation! 

While I understand that v7 alphas have improved DB2 support, we have concerns about using alpha versions in a production environment due to the potential instability and lack of long-term support. 

Could you clarify if there are any specific guarantees or assurances regarding the stability and readiness of v7 for production use? 

Additionally, we'd like to know if there's an estimated timeline for a stable v7 release.

Thank you!

**Comment by WikiRik** [comment]
> While I understand that v7 alphas have improved DB2 support, we have concerns about using alpha versions in a production environment due to the potential instability and lack of long-term support. 
> 
> Could you clarify if there are any specific guarantees or assurances regarding the stability and readiness of v7 for production use? 

No, that's why it's called an alpha release. We are basically there to release a beta with the same code but we want to be able to provide active support for the community while they migrate projects and third-party packages. At this time we don't have enough capacity amongst the maintainers for it so we hold off from releasing the first beta.

> Additionally, we'd like to know if there's an estimated timeline for a stable v7 release.

With what I mentioned before, I cannot give an estimated timeline for a stable v7 release at this stage.
